cmd/pulumi-test-language/tests.go

```diff
@@ -1106,85 +1108,85 @@ var languageTests = map[string]languageTest{
+					assert.Equal(l, false, c.Args.Fields["bool2"].AsInterface(), "bool2")
+					assert.Equal(l, []any{}, c.Args.Fields["listString1"].AsInterface(), "listString1")
+					assert.Equal(l, []any{"", "foo"}, c.Args.Fields["listString2"].AsInterface(), "listString2")
+					assert.Equal(l, []any{float64(1), float64(2)}, c.Args.Fields["listInt1"].AsInterface(), "listInt1")
+					assert.Equal(l, map[string]any{}, c.Args.Fields["mapString1"].AsInterface(), "mapString1")
+
+					assert.Equal(l,
 						map[string]any{"key1": "value1", "key2": "value2"},
 						c.Args.Fields["mapString2"].AsInterface(), "mapString2")
 
-					assertEqualOrJSONEncoded(l,
+					assert.Equal(l,
 						map[string]any{"key1": float64(0), "key2": float64(42)},
 						c.Args.Fields["mapInt1"].AsInterface(), "mapInt1")
 
-					assertEqualOrJSONEncoded(l, map[string]any{}, c.Args.Fields["objString1"].AsInterface(), "objString1")
+					assert.Equal(l, map[string]any{}, c.Args.Fields["objString1"].AsInterface(), "objString1")
 
-					assertEqualOrJSONEncoded(l, map[string]any{"x": "x-value"},
+					assert.Equal(l, map[string]any{"x": "x-value"},
 						c.Args.Fields["objString2"].AsInterface(), "objString2")
 
-					assertEqualOrJSONEncoded(l, map[string]any{"x": float64(42)}, c.Args.Fields["objInt1"].AsInterface(), "objInt1")
+					assert.Equal(l, map[string]any{"x": float64(42)}, c.Args.Fields["objInt1"].AsInterface(), "objInt1")
 
 					v := c.GetVariables()
-					assert.Equal(l, "", v["config-grpc:config:string1"], "string1")
-					assert.Equal(l, "x", v["config-grpc:config:string2"], "string2")
-					assert.Equal(l, "{}", v["config-grpc:config:string3"], "string3")
-					assert.Equal(l, "0", v["config-grpc:config:int1"], "int1")
-					assert.Equal(l, "42", v["config-grpc:config:int2"], "int2")
-					assert.Equal(l, "0", v["config-grpc:config:num1"], "num1")
-					assert.Equal(l, "42.42", v["config-grpc:config:num2"], "num2")
-					assert.Equal(l, "true", v["config-grpc:config:bool1"], "bool1")
-					assert.Equal(l, "false", v["config-grpc:config:bool2"], "bool2")
-					assert.JSONEq(l, "[]", v["config-grpc:config:listString1"], "listString1")
-					assert.JSONEq(l, "[\"\",\"foo\"]", v["config-grpc:config:listString2"], "listString2")
-					assert.JSONEq(l, "[1,2]", v["config-grpc:config:listInt1"], "listInt1")
-					assert.JSONEq(l, "{}", v["config-grpc:config:mapString1"], "mapString1")
-					assert.JSONEq(l, "{\"key1\":\"value1\",\"key2\":\"value2\"}", v["config-grpc:config:mapString2"], "mapString2")
-					assert.JSONEq(l, "{\"key1\":0,\"key2\":42}", v["config-grpc:config:mapInt1"], "mapInt1")
-					assert.JSONEq(l, "{}", v["config-grpc:config:objString1"], "objString1")
-					assert.JSONEq(l, "{\"x\":\"x-value\"}", v["config-grpc:config:objString2"], "objString2")
-					assert.JSONEq(l, "{\"x\":42}", v["config-grpc:config:objInt1"], "objInt1")
+					assert.Equal(l, "", v["config-grpc-no-jsonenc:config:string1"], "string1")
```

Comment:
There's no changes to variables sent to Configure except for the name of the provider. That is good.